### PR TITLE
add extension for dictionaries

### PR DESCRIPTION
**Pull request by lopste**
This extension allows you to create dictionaries, parse JSON into one for usage, and then stringify them back. (`vercte/dictionaries.js`)
List of blocks in the extension, which include modifying dictionaries and converting them to and from JSON

Thumbnail:
Thumbnail
(yes I completely copied lukemaniastudioss' pull request template I'm new here)

### CODE REVIEW

**Comment by OeildeLynx31** [comment]
What are differences between this extension and the JSON extension?

**Comment by LukeManiaStudioss** [comment]
This looks really cool! 👌👌

**Comment by lopste** [comment]
>
> What are differences between this extension and the JSON extension?
>
This contains support for actively writing to the object and easily
retrieving keys from it, along with incrementing keys. Eventually, I’ll add
support for nested objects and arrays. It also internally uses maps to make
it more secure (but I’m not too sure how far you can breach into it using
Scratch as a front-end.)

**Comment by OeildeLynx31** [comment]
Ok, thank you!

File: extensions/vercte/dictionaries.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
+window.dictionaries = new Map();
```
Comment (GarboMuffin) [inline extensions/vercte/dictionaries.js:1]: in addition to all the other warnings, extensions cannot store things on `window`; use the IIFE from https://docs.turbowarp.org/development/extensions/unsandboxed then just define `var dictionaries = `

File: extensions/vercte/dictionaries.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+class DictionaryExtension {
+  getInfo() {
+    return {
+      id: 'verctedictionaries',
+      name: 'Dictionaries',
```
Comment (GarboMuffin) [inline extensions/vercte/dictionaries.js:6]: Also, since your image has a well defined color scheme, you should set the block color accordingly

**Comment by lopste** [comment]
Also, looks like the lint failed. I swear I linted it, hold on.

**Comment by lopste** [comment]
I fixed all the issues mentioned and added some useful new blocks since keys couldn't be easily removed.

**Comment by GarboMuffin** [comment]
do you consider this extension to be done or are you adding more changes

**Comment by lopste** [comment]
I think it's good for not. I may add more changes in the feature, specifically nested objects, but it's done for now.

File: extensions/vercte/dictionaries.js [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+  }
+
+  dict_delete_key({ KEY, DICT }) {
+    if (dictionaries.has(DICT)) {
+      dictionaries.get(DICT).delete(KEY);
+    }
+  }
+
+  dict_primitive_null() {
+ return null;
```
Comment (GarboMuffin) [inline extensions/vercte/dictionaries.js:192]: blocks can't return null. unknown things break when that happens

**Comment by lopste** [comment]
Oh. Hm. I guess I should just remove that block and add an option to
nullify a property instead. Or just bring null out of the equation.

**Comment by LukeManiaStudioss** [comment]
> Does it need an icon? Screenshot of "Dictionaries" extension in editor

It's not required, no

**Pull request merged**